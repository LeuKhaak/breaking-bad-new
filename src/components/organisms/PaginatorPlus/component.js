import React from "react";
import * as styles from "./styles.module.scss";
import Paginator from "../../../components/organisms/Paginator";
import ForPaginator from "../../../components/molecules/ForPaginator";

function PaginatorPlus({ loader, perPage, amounts, changeLimitPages }) {
  return (
    <div className={styles.container} style={loader ? { display: "none" } : {}}>
      <Paginator />
      <ForPaginator
        loader={loader}
        perPage={perPage}
        amounts={amounts}
        changeLimitPages={changeLimitPages}
      />
    </div>
  );
}

export default PaginatorPlus;
