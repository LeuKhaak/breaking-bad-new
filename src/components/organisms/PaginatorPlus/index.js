export * from "./component";
export { container as default } from "./container";
