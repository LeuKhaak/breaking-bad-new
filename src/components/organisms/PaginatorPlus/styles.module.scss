@import "../../../assets/styles/styles.scss";
.container {
  @include content;
  max-width: 74.5rem;
  display: flex;
  justify-content: space-between;
  margin-top: 3.5rem;
  @include mobile {
    flex-direction: column-reverse;
    align-items: center;
  }
}
