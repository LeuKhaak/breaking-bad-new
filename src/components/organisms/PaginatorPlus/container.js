import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { changeLimit } from "src/store/actions/paginator";
import PaginatorPlus from "./component";

function PaginatorPlusContainer() {
  const loader = useSelector((state) => state.cards.loader);
  const dispatch = useDispatch();
  const perPage = useSelector((state) => state.paginator.perPage);
  const amounts = [5, 10, 15, 20];

  const changeL = (val) => {
    dispatch(changeLimit(val));
  };
  return (
    <PaginatorPlus
      loader={loader}
      perPage={perPage}
      amounts={amounts}
      changeL={changeL}
    />
  );
}

export const container = PaginatorPlusContainer;
