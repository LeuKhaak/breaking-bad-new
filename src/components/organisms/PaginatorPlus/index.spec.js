import React from "react";
import { shallow } from "enzyme";
import PaginatorPlus from "./component";

let component = null;
const setComponent = (props) => shallow(<PaginatorPlus {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("PaginatorPlus testing:", () => {
  test("is PaginatorPlus not rendered", () => {
    component = setComponent({
      loader: true,
    });
    expect(component).toMatchSnapshot();
  });
  test("is PaginatorPlus rendered", () => {
    component = setComponent({
      loader: false,
    });
    expect(component).toMatchSnapshot();
  });
});
