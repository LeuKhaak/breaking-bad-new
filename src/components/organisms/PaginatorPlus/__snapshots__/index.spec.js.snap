// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatorPlus testing: is PaginatorPlus not rendered 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
>
  <PaginatorContainer />
  <ForPaginator
    loader={true}
  />
</div>
`;

exports[`PaginatorPlus testing: is PaginatorPlus rendered 1`] = `
<div
  style={Object {}}
>
  <PaginatorContainer />
  <ForPaginator
    loader={false}
  />
</div>
`;
